ide/parsing.lucene/src/org/netbeans/modules/parsing/lucene/MemoryIndex.java

```diff
@@ -304,7 +290,7 @@ private synchronized IndexReader getReader() throws IOException {
     private synchronized void refreshReader() throws IOException {
         assert lock.isWriteLockedByCurrentThread();
         if (cachedReader != null) {
-            final IndexReader newReader = cachedReader.reopen();
+            IndexReader newReader = IndexReader.openIfChanged(cachedReader);
```

Comment:
According to my reading of the documentation this is not correct. The two functions have different semantics:

 `openIfChanged` says:

> Returns:
>   null if there are no changes; else, a new IndexReader instance which you must eventually close 

While `reopen` says:

>  If the index has not changed since this instance was (re)opened, then this call is a NOOP and returns this instance. Otherwise, a new instance is returned. The old instance is not closed and remains usable.

Reply:
good catch will fix.
`reopen()` is implemented in 3.6.2 as:
```
    final IndexReader newReader = IndexReader.openIfChanged(this);
    if (newReader == null) {
      return this;
    } else {
      return newReader;
    }
```
will inline this. 4.0.0 moved `openIfChanged` to `DirectoryReader` but it still exists, so it is a useful refactoring still.